### PR TITLE
#263 - Artworks for sale indicator

### LINKED ISSUE #263: Artwork for sale indicator
Make it easier for users to understand and determine when an artwork is available for sale.

Screen Shot 2024-05-23 at 8 09 13 PM

Screen Shot 2024-05-23 at 8 09 20 PM

### PR DESCRIPTION
**Pull request by bexeGrace**
Closes #263

### CODE REVIEW

**Review by MAbdurrehman1 — CHANGES_REQUESTED** [review]
Can we please do the following:

1. Add some space here: 
image

2. Make this a bit smaller as shown in the design in issue ticket
image

3. Combine these in 1 row, as shown in the design
image

4. Make the button label as same in the design
image

5. On the artworks on profile page, can we have some sort of differentiator/indicator that which artwork is in draft and which artwork is ready to be sold? I think we can use a badge here. (we already have a component for it)
image

6. Important one: If I have made an artwork ready to be sold on marketplace, and I edit it and save it as draft, it still shows up on the marketplace
image
image
image

Solution on my mind: Currently, we do not have any flag in artwork model that can decide whether an artwork is in draft or not, since we are deciding that on whether the currency and amount are provided or not. I think this should be fine for now, but IF an artwork already has currency and amount then we don't show the toggle buttons to make it draft. OR we can set the currency and amount to null if user wants to draft the artwork again.

**Review by bexeGrace — COMMENTED** [review]
Modified the required ui changes and added the badges on the artworks in profile page
Screenshot (82)
Screenshot (83)

**Comment by bexeGrace** [comment]
removed the badges in the marketplace page and modified the UI in the following manner
Screenshot (85)
 manner

**Review by MAbdurrehman1 — CHANGES_REQUESTED** [review]
Can we have these label in a single line? instead of going in next line.

image

Whenever I edit an artwork from my profile, it just has 1 tab showing with 2 input fields, which I think is confusing for users.
image
I think what we want to do here is: If an artwork is already in draft, then users still have the tabs options working but if an artwork is ready to be sold, and then we can hide the tabs options and show all fields (basically, we are not allowing users to make an active artwork as draft again). Does this makes sense?

The badges are showing on top of edit modals:
image

File: src/components/ArtworkCard/index.tsx [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
+import Badge, {BadgeStyle} from 'components/Badge';
+import Price from 'components/Price';
 
 export interface ArtworkCardProps {
   artwork: Artwork;
+  pageName: string;
 }
 
-const ArtworkCard: SFC<ArtworkCardProps> = ({artwork, className}) => {
+const ArtworkCard: SFC<ArtworkCardProps> = ({artwork, className, pageName}) => {
```
[the remark this answers is not in the document]
Comment (bexeGrace) [reply src/components/ArtworkCard/index.tsx:19]: Okay

File: src/components/ArtworkCard/Styles.ts [hunk 2]
```diff
@@ -18,6 +17,7 @@ export const Container = styled.div`
   box-shadow: 0 3px 6px rgb(140 149 159 / 15%);
   overflow: hidden;
   transition: all 0.2s ease-in-out 0s;
+  position: relative; /* Ensure the container is positioned relative */
```
[the remark this answers is not in the document]
Comment (bexeGrace) [reply src/components/ArtworkCard/Styles.ts:20]: Alright

File: src/components/ArtworkCard/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bexeGrace) [reply src/components/ArtworkCard/index.tsx:19]: Modified it to be showBadge which accepts boolean value

**Review by MAbdurrehman1 — APPROVED** [review]
👍

**Pull request merged**